static/js/tools/choropleth/choropleth_template.tsx

```diff
@@ -32,13 +36,138 @@ window.onload = () => {
 };
 
 class MainPane extends Component {
+  constructor(props: Record<string, unknown>) {
+    super(props);
+
+    // Redirect to basic url if none provided.
+    if (window.location.search === "") {
+      window.location.search =
+        "statVar=Count_Person_Employed&pc=t&geoDcid=country/USA";
+    }
+
+    // Bind functions as needed.
+    this._handleStatVarSelection = this._handleStatVarSelection.bind(this);
+
+    // Get default values for optional fields.
+    const urlParams = new URLSearchParams(window.location.search);
+    let isPerCapita = false;
+    if (urlParams.has("pc")) {
+      isPerCapita = ["true", "t", "1"].includes(
+        urlParams.get("pc").toLowerCase()
+      );
+    }
+
+    // Get all statistical variable available for the current subgeo.
+    axios
+      .get("/api/place/child/statvars/" + urlParams.get("geoDcid"))
+      .then((resp) => {
+        if (resp.status === 200) {
+          const statVars: Set<string> = new Set();
+          resp.data.forEach((item) => statVars.add(item));
+          this.setState({
+            statsVarFilter: new TimelineStatsVarFilter(statVars),
+          });
+        }
+      });
+
+    // Initialize state.
+    this.state = {
+      // References used for both choropleth and stats var sidemenu objects.
+      choroplethMap: React.createRef(),
+      statVarMenuRef: React.createRef(),
+      pc: isPerCapita,
+      // Default to no filtering in stats var sidemenu until API call returns.
+      statsVarFilter: new NoopStatsVarFilter(),
+      // Tracks the currently selected node in sidemenu.
+      statsVarNodes: {},
+    };
+  }
+
+  /**
+   * Toggles the per capita value of the choropleth map and redraws the map.
+   */
+  _togglePerCapita(): void {
+    // Update locally in parent.
+    const newPerCapitaValue = !this.state["pc"];
+    this.setState({
+      pc: newPerCapitaValue,
+    });
+    // Redraw the choropleth map.
+    const choroplethRef = this.state["choroplethMap"].current;
+    choroplethRef.setPerCapita(newPerCapitaValue);
+    choroplethRef.updateGeoValues();
+  
+    // Update in URL.
+    const urlParams = new URLSearchParams(window.location.search);
+    urlParams.set("pc", newPerCapitaValue.toString());
+    history.pushState({}, null, "choropleth?" + urlParams.toString());
```

Comment:
Would be best to centralize updating the history stack, or at least use the helper.

Reply:
Added a todo

Reply:
Did you forget to add the todo, or upload a commit?